agent/consul/connect_ca_endpoint.go

```diff
@@ -217,6 +217,11 @@ func (s *ConnectCA) Roots(
 		return err
 	}
 
+	// Exit early if Connect hasn't been enabled.
+	if !s.srv.config.ConnectEnabled {
+		return ErrConnectNotEnabled
+	}
```

Comment:
This seems important enough that it should have a unit test, also easy to test.

Reply:
It does - they are all the ones that are broken right now as they asserted the old behaviour! Will be fixed soon.